provider/script.go

```diff
@@ -0,0 +1,91 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func scriptResource() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this resource to run a script from an agent.",
+		CreateContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+			rd.SetId(uuid.NewString())
+			return nil
+		},
+		ReadContext:   schema.NoopContext,
+		DeleteContext: schema.NoopContext,
+		Schema: map[string]*schema.Schema{
+			"agent_id": {
+				Type:        schema.TypeString,
+				Description: `The "id" property of a "coder_agent" resource to associate with.`,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the script to display logs in the dashboard.",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"log_path": {
+				Type:        schema.TypeString,
+				Description: "The path of a file to write the logs to. If relative, it will be appended to tmp.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"icon": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Description: "A URL to an icon that will display in the dashboard. View built-in " +
+					"icons here: https://github.com/coder/coder/tree/main/site/static/icon. Use a " +
+					"built-in icon with `data.coder_workspace.me.access_url + \"/icon/<path>\"`.",
+			},
+			"source": {
```

Comment:
The docs say `scripts` but this says `source`. I prefer `script` tbh, especially if it's supposed to be an inline script as opposed to an existing file.

Reply:
Hmm, fair. Source could mean like GitHub or something wack, I'll change it back